### PR TITLE
Fix/correct auth flow

### PR DESCRIPTION
**Pull request by VitaliiDenysiuk**

### CODE REVIEW

File: src/redux/reducers/user.ts [hunk 1]
```diff
@@ -127,16 +130,18 @@ const userSlice = createSlice({
     });
     builder.addCase(registerUser.fulfilled, (store, { payload }) => {
       store.error = undefined;
-      return {
-        ...store,
+      localStorage.setItem("accessToken", payload.accessToken);
+      localStorage.setItem("refreshToken", payload.refreshToken);
+      Object.assign(store, {
+        ...userInitialState,
         ...payload.user,
```
Comment (alexandrtovmach) [inline src/redux/reducers/user.ts:137]: Better to use return with spread approach, instead of Object.assign

File: src/redux/reducers/user.ts [hunk 2]
```diff
@@ -148,16 +153,18 @@ const userSlice = createSlice({
     });
     builder.addCase(loginUser.fulfilled, (store, { payload }) => {
       store.error = undefined;
-      return {
-        ...store,
+      localStorage.setItem("accessToken", payload.accessToken);
+      localStorage.setItem("refreshToken", payload.refreshToken);
+      Object.assign(store, {
```
Comment (alexandrtovmach) [inline src/redux/reducers/user.ts:158]: Same

File: src/redux/reducers/user.ts [hunk 3]
```diff
@@ -116,6 +116,9 @@ const userSlice = createSlice({
     addAvatar: (state, action: PayloadAction<string>) => {
       state.profile.avatar = action.payload;
     },
+    cleanError: (state) => {
```
Comment (alexandrtovmach) [inline src/redux/reducers/user.ts]:
```suggestion
    clearError: (state) => {
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

File: src/redux/reducers/user.ts (hunk 1 above)
Comment (VitaliiDenysiuk) [reply src/redux/reducers/user.ts:137]: spread did not save user to store

File: src/redux/reducers/user.ts (hunk 2 above)
Comment (VitaliiDenysiuk) [reply src/redux/reducers/user.ts:158]: spread did not save user to store

**Review by DavidKorystov — APPROVED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**